### PR TITLE
53217: Improve Redactor Custom Resource Content

### PR DESCRIPTION
**Pull request by jonquil2002**
SC: https://app.shortcut.com/replicated/story/53217/better-reference-content-on-redactor-custom-resources

Preview: https://deploy-preview-507--replicated-docs.netlify.app/reference/custom-resource-redactor

### CODE REVIEW

File: docs/reference/custom-resource-redactor.md [hunk 1]
```diff
@@ -18,3 +16,76 @@ metadata:
 spec:
   redactors: []
 ```
+
+## Objects and Fields
+
+Each redactor must contain the `fileSelector` and `removals` objects.
```
Comment (divolgin) [inline docs/reference/custom-resource-redactor.md:22]: If `fileSelector` is omitted, the redactor will be applied to all files: https://github.com/replicatedhq/troubleshoot/blob/ff5ee2b709948712120cbf75aa5e940bd0b9f664/pkg/redact/redact.go#L129-L131

File: docs/reference/custom-resource-redactor.md [hunk 2]
```diff
@@ -18,3 +16,76 @@ metadata:
...
+    <th width="30%">Object or Field Name</th>
+    <th width="70%">Description</th>
+  </tr>
+  <tr>
+    <td><code>fileSelector</code></td>
+    <td>(Required) This object determines which files the redactor is applied to.</td>
+  </tr>
+  <tr>
+    <td><code>removals</code></td>
+    <td>(Required) This object determines which files are removed.</td>
```
Comment (divolgin) [inline docs/reference/custom-resource-redactor.md:46]: It determines which strings in each file are redacted.

File: docs/reference/custom-resource-redactor.md [hunk 3]
```diff
@@ -18,3 +16,76 @@ metadata:
...
+- Set of files that the redactor can apply to
+- Set of string literals to replace
+- Set of regex replacements to run
+- List of YAML paths to redact
+
+Any of the four fields can be omitted. For more information and examples of these fields, see [Redactors](https://troubleshoot.sh/docs/redact/redactors/) in the Troubleshoot documentation.
+
+The following objects and fields are used to define custom redactors:
+
+<table>
```
Comment (divolgin) [inline docs/reference/custom-resource-redactor.md:35]: This table shows a flat structure, but `file` and `files` are nested under `fileSelector`, for example.

`values`, `regex`, and `yamlPath` are under `removals`

**Review by jonquil2002 — COMMENTED** [review]
lgtm, but I can't approve.

**Comment by ajp-io** [comment]
Thanks, I'll approve and merge in a sec.

**Pull request merged**